autopts/bot/common_features/report.py

```diff
@@ -68,106 +69,176 @@ def make_repo_status(repos_info):
-    :param descriptions: test cases
-    :return:
+    :param progresses_list: list of progresses found
+    :param descriptions: dictionary with test cases descriptions
+    :param xmls: directory where XML files are located
+    :param errata: dictionary with errata notes for specific test cases
+    :return: None
     """
 
     try:
         xml_list = list(os.scandir(xmls))
-    except FileNotFoundError as e:
-        log("No XMLs found")
+    except FileNotFoundError:
+        print("No XMLs found")
         xml_list = None
-    matched_xml = ''
-
-    def find_xml_by_case(case):
-        if xml_list is None:
-            return
-        nonlocal matched_xml
-        matched_xml = ''
-        to_match = case.replace('/', '_').replace('-', '_')
-        for xml in xml_list:
-            if to_match in xml.name:
-                matched_xml = xml.name
-                break
 
-    header = "AutoPTS Report: " \
-             "{}".format(datetime.datetime.now().strftime("%Y-%m-%d %H:%M"))
+    header = "AutoPTS Report: {}".format(datetime.datetime.now().strftime("%Y-%m-%d %H:%M"))
     workbook = xlsxwriter.Workbook(report_xlsx_path)
-    worksheet = workbook.add_worksheet()
-    chart = workbook.add_chart({'type': 'pie',
-                                'subtype': 'percent_stacked'})
+    worksheet = workbook.add_worksheet("Report")
+    chart = workbook.add_chart({'type': 'pie', 'subtype': 'percent_stacked'})
 
-    # Add a bold format to use to highlight cells.
-    bold = workbook.add_format({'bold': True})
+    bold_format = workbook.add_format({'bold': True})
 
-    # Write data headers.
     worksheet.write('A1', header)
-    worksheet.write_row('A3', ['Test Case', 'Result', 'XML'])
-
-    row = 3
-    col = 0
-
-    for k, v in list(results_dict.items()):
-        worksheet.write(row, col, k)
-        if v[0] == 'PASS':
-            find_xml_by_case(k)
-            worksheet.write(row, col + 2, matched_xml)
-        if v[0] == 'PASS' and int(v[1]) > 1:
-            v = '{} ({})'.format(v[0], v[1])
-        else:
-            v = v[0]
-        if k in errata:
-            v += ' - ERRATA ' + errata[k]
-        worksheet.write(row, col + 1, v)
-        if k in list(descriptions.keys()):
-            worksheet.write(row, col + 3, descriptions[k])
-        if k in regressions_list:
-            worksheet.write(row, col + 4, "REGRESSION")
-        if k in progresses_list:
-            worksheet.write(row, col + 4, "PROGRESS")
-        row += 1
+    worksheet.write_row('A3', [
+        'Test Case',
+        'Result',
+        'Result Additional Info',
+        'XML',
+        'Description',
+        'Start Time',
+        'End Time',
+        'Duration'
+    ])
+
+    current_row = 3
+    start_col = 0
+
+    for test_case, result_data in results_dict.items():
+        raw_result = result_data[0]
+        parsed_result, additional_info = parse_result_column(raw_result)
+
+        if parsed_result == 'PASS' and len(result_data) > 1:
```

Comment:
I see you checking `len(result_data)` here and there. It seems that `result_data` is a tuple made at:
```python
results[tc_xml.attrib["name"]] = (status, run_count, test_start_time, test_end_time, duration)
```
Tuples in Python are immutable, meaning their size and contents cannot be changed after creation. If it is needed for the result_data to be updated at some point, I suggest to use a dictionary instead, so we don't have to wonder which index contains what. We could have a key like "additional_info" too.